File: app/src/main/java/com/wind/ndk/png/MainActivity.java
package com.wind.ndk.png;

import android.Manifest;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;

public class MainActivity extends Activity {


	static {
		System.loadLibrary("native");
	}
	private Button btn;
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		btn = (Button) findViewById(R.id.btn);
		btn.setOnClickListener(new OnClickListener() {
			
			@Override
			public void onClick(View v) {
				Intent intent = new Intent(MainActivity.this, PngActivity.class);
				startActivity(intent);
			}
		});
//内置sd卡路径

		requestPermissions(new String[]{Manifest.permission.READ_EXTERNAL_STORAGE,Manifest.permission.WRITE_EXTERNAL_STORAGE},1);
	}

}

File: app/src/main/java/com/wind/ndk/png/MainActivity.kt
package com.wind.ndk.png

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.SurfaceHolder
import android.view.SurfaceHolder.Callback
import android.widget.TextView
import com.wind.ndk.png.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var pngPlayer: PngPlayer
    private val pngPath="/mnt/sdcard/1.png"
    private val callback=object :Callback{
        override fun surfaceCreated(holder: SurfaceHolder) {
            pngPlayer=PngPlayer().apply {
                setDataSource(pngPath)
                setSurface(holder.surface)
            }

        }

        override fun surfaceChanged(holder: SurfaceHolder, format: Int, w: Int, h: Int) {
            pngPlayer.resize(w,h)
        }

        override fun surfaceDestroyed(holder: SurfaceHolder) {
        }

    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

       binding.surfaceView.holder.apply {
           addCallback(callback)
           //setType(SurfaceHolder.SURFACE_TYPE_PUSH_BUFFERS)
       }

    }


}